### DISCUSSION TITLE
Lesson 7 Part 4 FundMeTest.t.sol

### DISCUSSION THREAD

**Discussion by snowieoxs** [Cyfrin/foundry-full-course-cu#1580] [Q&A]
Hello!
[link to video](https://www.youtube.com/watch?v=eu3Wu9PcsW0&list=PL2-Nvp2Kn0FPH2xU3IbKrrkae-VVXs1vk&index=84)
So in the contract **FundMeTest.t.sol** I am having a issue compiling, and vscode is telling me it's this function:

```solidity
    function setUp() external {
        fundMe = new FundMe();
    }
```

Specifically It's saying that ```new FundMe()``` needs an argument, copilot is saying it needs ```fundMe = new FundMe(priceFeedAddress);``` with ```priceFeedAddress``` the address of my deployed  AggregatorV3Interface contract.

This is the code I have for **FundMeTest.t.sol** from following the video:

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../src/FundMe.sol";


contract FundMeTest is Test {
    FundMe fundMe;

    function setUp() external {
        fundMe = new FundMe();
    }

    function testMinimumDollarIsFive() public {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }
}
```

I tried adding the lines
```import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";```
added ```FundMe fundMe;``` as a state variable, so now it looks like this:

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../src/FundMe.sol";
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

contract FundMeTest is Test {
    FundMe fundMe;
    AggregatorV3Interface priceFeed;

    function setUp() external {
        fundMe = new FundMe(priceFeed);
    }

    function testMinimumDollarIsFive() public {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }
}
```

but... that doesn't work... I don't think that's how you get the address, and I'm not even sure that is what it really needs!

Thank you!

**Comment by EngrPips** [comment]
Hello @snowieoxs , from the look of things it appears to me like you are trying to build the codebase yourself and not following what Patrick is doing in the video. You are suppose to have a `DeployFundMe` script, please show your `DeployFundMe` script.

**Reply by snowieoxs** [reply]
So, I restarted Lesson 7 and Copied the **FundMe.sol** and **PriceConverter.sol** contract from here [https://github.com/Cyfrin/remix-fund-me-f23/blob/main/PriceConverter.sol](https://github.com/Cyfrin/remix-fund-me-f23/blob/main/PriceConverter.sol) and it's working now.

Patrick did say to grab the contract from the lesson 4 codebase, and I had grabbed it from somewhere else, in the repo.

Thanks, EngrPips!

**Reply by EngrPips** [reply]
Awesome, I am glad you are able to figure it out. keep crushing it friend

**Comment by neo-t-maredi** [comment]
Hey Guys so my errors are seemingly all compilation based: 

 error="/root/foundry-fund-me-f24/lib/forge-std/src/src/Test.sol": No such file or directory (os error 2)
Error: 
failed to resolve file: "/root/foundry-fund-me-f24/lib/forge-std/src/src/Test.sol": No such file or directory (os error 2); check configured remappings
        --> /root/foundry-fund-me-f24/test/FundMeTest.t.sol
![Screenshot 2024-04-12 142729](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/76637ed8-f48f-4881-8766-a282dcd08e96)
![Screenshot 2024-04-12 142741](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/95e36c28-87cc-4d8c-9edc-808be55640ac)
![Screenshot 2024-04-12 142758](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/a2246f84-d64d-428e-8c9c-236af8608e01)

Any help would be greatly appreciated. It's been infuriating to say the least

**Reply by EngrPips** [reply]
Please open the `lib` and confirm that there are stuffs inside of it.

**Reply by neo-t-maredi** [reply]
Hey there. thanks for getting back to me :) 

Alright so I restarted the entire thing and had to add some new dependencies etc and now it's throwing this error
![Screenshot 2024-04-12 211316](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/a68a68ca-7748-44c0-b3f9-77b21824ee05)
![Screenshot 2024-04-12 211339](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/a7aca9b4-936e-4684-96a0-1b736c5b8e5e)
![Screenshot 2024-04-12 211405](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/0c7db264-0909-4839-aa5c-4c9b56417d09)

Thoughts? Is it a remapping issue?

**Reply by EngrPips** [reply]
Hey @neo-t-maredi , did you ensure to download the exact version of the `chainlink` repo Patrick downloaded in the video tutorial ?, because any other version other than what Patrick downloaded in the tutorial might lead to you not have the file in the expected path or not even having it at all.

**Reply by neo-t-maredi** [reply]
Hey Buddy!

Yeah I did and nothing is seemingly working. Restarted this project twice

**Reply by EngrPips** [reply]
Hmmmmm, that seem very weird. I mean it shouldn't be that difficult. maybe push your code to github and share me the link so i can try to figure out what is wrong locally on my machine

**Reply by neo-t-maredi** [reply]
Christ Alive, finally figured it out. After a system overhaul and refresh we're cooking with grease :)

**Reply by EngrPips** [reply]
Awesome, i am glad you get it to work finally. keep crushing it!!!
